### PR TITLE
Indent internal records type definitions.

### LINKED ISSUE #13756: Signature file generation does not handle internal record with xml comment
When generating a signature file from an internal record with a code comment, the generated code is not indented far enough.

**Repro steps**

```fsharp
module SignatureFileGeneration.MyModule

type PullActions =
    internal
        {
            /// Any repo which doesn't have a master branch will have one created for it.
            Log : int
        }
```

leads to

```fsharp
module SignatureFileGeneration.MyModule

type PullActions =
    internal
    {
      
      /// Any repo which doesn't have a master branch will have one created for it.
      Log: int
    }
```

**Expected behaviour**

The generated signature should be considered equivalent to the backing source file.

**Actual behaviour**

The generated code isn't valid:

image

**Known workarounds**

Edit signature file by hand.

### PR DESCRIPTION
**Pull request by nojaf**
Hello @dsyme, in this PR I tried to fix #13756.
The original problem was discovered using `--allsigs`, which squashes every layout to 80.

https://github.com/dotnet/fsharp/blob/8385830e1f36e5005f952bd1267b969922d0b20f/src/Compiler/Driver/fsc.fs#L341-L347

That is why I've extended `FSharpCheckerResults.GenerateSignature` with an optional page width parameter. I hope that is ok.

### CODE REVIEW

**Review by KevinRansom — APPROVED** [review]
It took a little thinking about how this solved the problem, but then I got it :-)

Nice,

thanks

**Review by vzarytovskii — APPROVED** [review]

File: tests/FSharp.Test.Utilities/Compiler.fs [hunk 1]
```diff
@@ -1331,13 +1331,15 @@ module rec Compiler =
...
         |> string
         |> fun s ->
             s.Replace("\r", "").Split('\n')
             |> Array.map (fun line -> line.TrimEnd())
             |> String.concat "\n"
-            |> fun tap -> tap
\ No newline at end of file
+    
+    let printSignatures cUnit = printSignaturesImpl None cUnit
+    let printSignaturesWith pageWidth cUnit = printSignaturesImpl (Some pageWidth) cUnit
```
Comment (0101) [inline tests/FSharp.Test.Utilities/Compiler.fs:1345]: Maybe this should be named `printSignaturesWithPageWidth` or `printSignaturesWithWidth` so it's more obvious at the call site

**Review by 0101 — APPROVED** [review]

File: tests/FSharp.Test.Utilities/Compiler.fs (hunk 1 above)
Comment (nojaf) [reply tests/FSharp.Test.Utilities/Compiler.fs:1345]: I can do this in a future PR. I'd like to have this one merged in as it contains some infrastructure that will be useful to tackle the next slew of problems.

Comment (0101) [reply tests/FSharp.Test.Utilities/Compiler.fs:1345]: Sure, it's not critical :)

**Pull request merged**